lambda-natchez/src/main/scala/feral/lambda/natchez/TracedLambda.scala

```diff
@@ -16,18 +16,26 @@
 
 package feral.lambda.natchez
 
+import cats.Monad
 import cats.effect.kernel.MonadCancelThrow
+import cats.syntax.all._
 import feral.lambda.Lambda
 import natchez.EntryPoint
 import natchez.Trace
 
 object TracedLambda {
-  def apply[F[_]: MonadCancelThrow, G[_], Event: HasKernel, Result](entryPoint: EntryPoint[F])(
-      lambda: Trace[G] => Lambda[G, Event, Result])(
-      implicit lift: LiftTrace[F, G]): Lambda[F, Event, Result] = { (event, context) =>
+  def apply[F[_]: MonadCancelThrow, G[_]: Monad, Event: HasKernel, Result](
+      entryPoint: EntryPoint[F])(
+      lambda: Trace[G] => Lambda[G, Event, Result]
+  )(implicit lift: LiftTrace[F, G]): Lambda[F, Event, Result] = { (event, context) =>
     val kernel = HasKernel[Event].extract(event)
     entryPoint.continueOrElseRoot(context.functionName, kernel).use { span =>
-      lift.run(span)(lambda(_)(event, context.mapK(lift.liftK)))
+      lift.run(span) { trace =>
+        trace.put(
+          Tags.arn(context.invokedFunctionArn),
+          Tags.requestId(context.awsRequestId)
+        ) >> lambda(trace)(event, context.mapK(lift.liftK))
```

Comment:
Generally I prefer composability to hard-coding, but this doesn't seem unreasonable?

Reply:
Except, should this be starting a new span or something? I don't really have experience with tracing 😕

Reply:
`continueOrElseRoot` starts a new span inside the root span from which it's continuing, so I think this should be ok.

Reply:
Thanks, good to know!